code/modules/multiz/basic.dm

```diff
@@ -33,4 +33,25 @@ var/global/list/z_levels = list() // Each Z-level is associated with the relevan
 	else if (dir == DOWN)
 		. = GetBelow(ref)?.resolve_to_actual_turf()
 	else
-		. = get_step_resolving_mimic(ref, dir)
\ No newline at end of file
+		. = get_step_resolving_mimic(ref, dir)
+
+/proc/get_dir_multiz(turf/us, turf/them)
+	us = get_turf(us)
+	them = get_turf(them)
+	if(!us || !them)
+		return 0
+	if(us.z == them.z)
+		return get_dir(us, them)
+	else
+		var/turf/T = GET_ABOVE(us)
```

Comment:
Maybe this can be optimized a little by first comparing z levels and then getting above/below turf.
something like if them.z > us.z && get_above(us) and vice versa for below